pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,441 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+import "k8s.io/api/core/v1/generated.proto";
+import "k8s.io/api/autoscaling/v1/generated.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/gogo/protobuf/protobuf/google/protobuf/wrappers.proto";
+
+message IstioInstaller {
+    InstallerSpec spec = 3;
+    InstallerStatus status = 4;
+}
+
+// InstallerSpec defines the desired state of IstioInstaller.
+message InstallerSpec {
+    // Prefix added to all namespaces for any installed component.
+    string default_namespace_prefix = 5;
+
+    // Selection and configuration of core Istio features.
+    TrafficManagementFeatureSpec traffic_management = 25;
+    PolicyTelemetryFeatureSpec policy_telemetry = 26;
+    SecurityFeatureSpec security = 28;
+    ConfigManagementFeatureSpec config_management = 29;
+
+    // Ingress/egress gateway installation and configuration.
+    repeated IngressGatewaySpec ingress_gateway = 30;
+    repeated EgressGatewaySpec egress_gateway = 31;
+
+    // List of external operators that will be installed (but not managed)
+    // by the IstioInstaller e.g. prometheus, grafana, kiali.
+    repeated OperatorSpec external_operators = 50;
+
+    // Global k8s defaults. Can be overridden per component.
+    KubernetesResourcesSpec k8s_defaults = 80;
+
+    // URL for the install package. e.g.
+    //   http://github.com/istio/istio/releases/install/1.1.2
+    //   http://github.com/istio/istio/releases/install/lts
+    //   file://tmp/istio-installer/nightly
+    // Uses compiled in install package by default.
+    string custom_package_path = 100;
+    // Root for docker image paths e.g. docker.io/istio-release.
+    // Releases are published to docker hub under 'istio' project.
+    // Daily builds from prow are on gcr.io, and nightly builds from circle on docker.io/istionightly
+    string hub = 101;
+    // Version tag for docker images e.g. 1.0.6
+    string tag = 102;
+}
+
+// Configuration options for traffic management.
+message TrafficManagementFeatureSpec {
+    // Selects whether traffic management is installed.
+    // Must be enabled to enable any sub-component.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all traffic management components are installed into.
+    string namespace = 2;
+
+    // Component specific config.
+    PilotComponentSpec pilot = 30;
+    ProxyComponentSpec proxy = 31;
+    SidecarInjectorComponentSpec sidecar_injector = 32;
+}
+
+// Configuration options for the policy and telemetry feature.
+message PolicyTelemetryFeatureSpec {
+    // Default behavior of the sidecar for handling outbound traffic from the application.
+    enum OutboundPolicy {
```

Comment:
why outbound policy for the mesh is configured under policy/telemetry feature spec?

Reply:
So firstly I have to contradict myself a bit about an earlier statement that all non-k8s Istio config would still go through values.yaml. I think even in that case there may be some very coarse "feature selection" knobs that may be good to expose, even though the user could set that directly through values.yaml. These are the sort of knobs that 80% of installs may consider changing and this seems like a good match for that. 
As to why it's PolicyTelemetry rather than just Policy, if these are split again it would go into Policy feature.

Reply:
Should this be part of the security spec?  Should it apply globally, or is it intended that certain components would override global defaults?

Reply:
Policy is related to security but is its own thing in Istio and traffic policy is a part of that. It's global in the sense that it applies to all sidecars in the mesh and I don't think it makes sense to override this in the installer at least. Istio does provide config knobs to shape policy with finer granularity.

Reply:
I think outbound policy should be in traffic management feature, after all configuring outbound policy is for traffic management, unless this outbound policy is specific to the mixer policy pod/service.

Reply:
Okay, my comment was directed toward the organization of security related concerns.  If users are more familiar with configuring this in the context of mixer policy, then having it here is probably best.  Given that, would it make more sense to change "security" to certificate management?  Other than certs, it seems security related features are configured in the context of policy.

Reply:
My point here is yes, it's traffic management and yes, it's security, but in this case this is policy setting. Policy feature covers aspects of security and traffic management. 
It does happen to be implemented in mixer policy pod/service, but even conceptually I think this belongs in the policy feature group.

Reply:
I think we should rename security above then, as it's only configuring certificate management.  That said, like galley, is it required to explicitly enable/disable citadel?  I suppose there are configuration details regarding how it integrates with the other components (e.g. SDS/node agents), but it seems like it's used by default for managing certs on the istio components, and additionally when mTLS is enabled, so does it make sense to have it exposed as its own component (other than for configuring how it's manifested in the install)?

Reply:
I think i'm strongly opposing to have outbound policy within PolicyFeatureSpec, because I could run Istio with only pilot and without mixer policy (or other policy feature) but wanted to configure my traffic outbound policy.  I think that is where people's heads are today, it'd be confusing to move to be some config under PolicyFeatureSpec.

Reply:
@ostromart, looking at those docs you linked, I think it would be beneficial to go through each of those sections and itemize the settings required for each scenario.  Personally, I think it would be great if the installer CRD schema aligned well with these, especially since the docs are organized conceptually around user tasks.  As I mentioned in another comment, for the new CRD schema, I don't think we should be focused on physical manifestation of these features (i.e. simply exposing the options available for each component/deployment), but should instead be focused on _what_ users want, not how what they want gets done.

Reply:
I think there's some overlap here and some decisions that need to be made.  The operator automates operation of the control plane, which, in my opinion, includes configuration.  Users should be capable of installing a control plane, configured to their specifications, without having to go through extra steps.  What is the usefulness if we don't do that?

Reply:
You can create a fully configured control plane by supplying the operator CR and a mesh configmap. The question is whether that configmap is its own thing or whether it's schematized and included in the operator CR and we're going with leaving it as its own thing for now. 
The main reason for that is if you embed it in the operator CRD, the components cannot read or respond to it directly. So to make the API dynamic, the operator would have to restart components whenever changes are made.

Reply:
I'm going to resolve this since the discussion is revolving around things that are missing rather than part of this PR. Let's tackle the question of adding more knobs to the operator spec separately - that will be a broader discussion and we need to involve other people.